### PR TITLE
Activities display

### PR DESCRIPTION
**Pull request by Dunn-Austen**
### What's this PR do?
- Creates the addActivity mutation & integrates it into onPress functionality within the Map component
- Amends various apiCalls to include additional properties in the returned data
- Builds out the 'agenda' portion of the Map component with various conditional rendering triggers
- Updates our state hooks and prop interactions between TripPreview and Map

### Where should the reviewer start?
- Thanks for the assist with the troubleshooting!

### Any background context you want to provide?
Next steps involve the following:
- Error handling regarding date selection in AddDestination 
- Polish styling decisions in the agenda portion of Map
- Strip out logs and comments

---

### Archived Troubleshooting
This PR adds the addActivity mutation to apiCalls & builds out the 'agenda' section of the Maps component. I'm stuck at the button click where the user presses 'yes' (yes to the whether they want to save a clicked activity to their trip).

I'm not sure of the error's origin. **It could be an issue with the addActivity code in apiCalls or with the handleYesClick logic in Maps, or something I'm missing entirely.**

It's mentioning something about an undefined value near or within the handleYesClick method, but also gives back the following
```
[Unhandled promise rejection: TypeError: undefined is not an object (near '...ordinates.longitude).then(function (resp...')]
```

To offer some context, the addActivity() mutation requires nine arguments. All of those are either being passed into Maps as props from TripsPreview, or being pulled from the clickedActivity state hook. This hook is set on click of a marker. Click the TEST TEST TEST button to fire console logs that give back the value for the clickedActivity and savedActvities states.

addActivity is imported from apiCalls and invoked within handleYesClick. 

**Another source of error is that I'm maybe not using aysnc await appropriately somewhere upstream.** handleFetchCLick is a method invoked within handleYesClick, and that runs the original getAllTrips query (passed from Trips). Maybe I need to toss async await on that variable/function in Trips?
https://inneka.com/programming/reactnative/possible-unhandled-promise-rejection-id0-warning-in-react-native/

Maybe I need returns in my .then()'s ?

**RESOLVED - DUPLICATED ADDACTIVITY METHODS IN MAP.JS**

### CODE REVIEW

File: Components/Maps/Maps.js [hunk 1]
```diff
@@ -35,11 +67,45 @@ export const Maps = ({ route }) => {
...
+
+  const handleYesClick = async () => {
+    await addActivity(dest.id, clickedActivity.title, endDate, clickedActivity.address, clickedActivity.description, clickedActivity.rating, clickedActivity.image, clickedActivity.coordinates.latitude, clickedActivity.coordinates.longitude)
+      .then(data => {
+        setSavedActivities([...savedActivities, data.data.createActivity.activity])
+        setClickedActivity('')
+        handleTripsFetch()
+      })
+      .catch(err => console.log(err))
+  }
```
Comment (sertmer) [inline Components/Maps/Maps.js]: Good work on re-rendering the agenda upon posting the saved activities. Looks nice all on one page

File: Components/Maps/Maps.js [hunk 2]
```diff
@@ -1,15 +1,31 @@
...
-  const [ activities, setActivities ] = useState(formattedMarkers)
-  const [ startDate, setStartDate ] = useState(dest.destination.startDate)
-  const [ endDate, setEndDate ] = useState(dest.destination.endDate)
-  const [ savedActivities, setSavedActivities ] = useState([])
+  const { dest, formattedMarkers, startDate, endDate, handleTripsFetch } = route.params
+  let [ activities, setActivities ] = useState(formattedMarkers)
+  let [ savedActivities, setSavedActivities ] = useState(dest.activitySet)
+  let [ clickedActivity, setClickedActivity ] = useState('')
+
+//Component is currently set up to rely on state that is populated by local actions NOT properly set up to re-render based off updated fetched data
```
Comment (ckoga) [inline Components/Maps/Maps.js]: We'll get rid this on the next branch

File: Components/Maps/Maps.js [hunk 3]
```diff
@@ -39,7 +56,44 @@ export const Maps = ({ route }) => {
...
+
+    setClickedActivity('')
+  }
+
+
+  const handleNoClick = () => {
+    setClickedActivity('')
+  }
+
+  const handleTest = () => {
```
Comment (ckoga) [inline Components/Maps/Maps.js]: Delete this as well.

File: apiCalls.js [hunk 4]
```diff
@@ -14,7 +14,7 @@ export const getActivities = (lat, long) => {
 
 export const getAllTrips = () => {
   const query = {
-      "query": "{allTrips(userApiKey: \"b9aead4b955bccb5c57ef830580f3de5\") {id name origin originAbbrev originLat originLong tripdestinationSet {destination {location abbrev lat long} startDate endDate activitySet {name date category}}}}"
+      "query": "{allTrips(userApiKey: \"b9aead4b955bccb5c57ef830580f3de5\") {id name origin originAbbrev originLat originLong tripdestinationSet {destination {location abbrev lat long} id startDate endDate activitySet {name date address category rating image lat long tripDestination { trip { name } destination { location abbrev } } } } } }"
```
Comment (ckoga) [inline apiCalls.js]: Still not sure whats happening here.

Comment (Dunn-Austen) [reply apiCalls.js]: A general overview:

Fetching from Graphql requires some adjustments to our typical RESTFUL fetching. All fetches use the POST method, and fall into two categories of queries (get data) and mutations (update, add, or delete data).

The second 'argument' of properties within the curlies (after the API key) tell the endpoint what to specifically return in the fetch response

**Pull request merged**